### ISSUE TITLE
Feature: variance thresholding from sklearn as an alternative to Adu filtering.

### ISSUE THREAD

**Issue by lg345** [lg345/XSpecT#36]
Example code 

from sklearn.feature_selection import VarianceThreshold
threshold = 0.01
selector = VarianceThreshold(threshold)

X_filtered = selector.fit_transform(X)

retained_indices = selector.get_support(indices=True)
retained_features = data.columns[retained_indices]

Then we would set all other indices to zero.

Not sure yet about the performance issues this might have. Seemed to work well in AMI-II

The goal being that the user would no longer need to think about the right/optimal ADU filtering.

**Comment by lg345** [comment]
Fred's intern Amine used this in AMI-II

**Comment by lg345** [comment]
Implementing as a new pipeline step `filter_detector_variance`, a drop-in alternative to `filter_detector_adu`.

**Approach**
- Flatten detector data to (shots, features) where features = flattened pixels.
- Fit `sklearn.feature_selection.VarianceThreshold(threshold=variance_threshold)` across shots.
- Build the retained-feature mask via `get_support()`, zero out all non-retained pixels, reshape back to the original detector shape.
- Write result back to the same detector key (in place, matching `filter_detector_adu`), and store the retained mask as `<key>_variance_mask` for diagnostics.

**YAML**
```yaml
- step: filter_detector_variance
  on: epix_ROI_1
  variance_threshold: 0.01
```

`variance_threshold` defaults to 0.0 (removes only constant/dead pixels). The point is to drop the hand-tuned ADU cutoff: the mask is data-driven from per-pixel variance across shots.

Handles both 3D (shots x rows x cols) and 2D (shots x pixels) detectors. Documented in the YAML guide; unit tests added (constant-pixel removal, threshold behavior, 2D/3D shapes, shape preservation).

**Comment by lg345** [comment]
Implemented in `filter_detector_variance` and merged into feature/yaml-pipeline-architecture (local). Registered step + unit tests + YAML guide docs. Not yet pushed to origin.
